pkg/storage/storage.go

```diff
@@ -142,6 +151,52 @@ func (s *Storage) History(name string) ([]*rspb.Release, error) {
 	return s.Driver.Query(map[string]string{"NAME": name, "OWNER": "TILLER"})
 }
 
+// removeLeastRecent removes items from history until the lengh number of releases
+// does not exceed max.
+//
+// We allow max to be set explicitly so that calling functions can "make space"
+// for the new records they are going to write.
+func (s *Storage) removeLeastRecent(name string, max int) error {
+	if max <= 0 {
+		return nil
+	}
+	h, err := s.History(name)
+	if err != nil {
+		s.Log("error fetching history: %s", err)
```

Comment:
- [ ] Don't both log and return errors.  
  Allow the caller to handle the error, given that this method already threatens to return an error.